### PR TITLE
[Data Virtualization]: Repro groupId and offline causing extra network calls

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#8937](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8937)

This PR is expected to fail until the fix is merged in: https://github.com/microsoft/FluidFramework/pull/21908

The repro is:
- Offline and Data virtualization are enabled
- A virtualized datastore is created
- It is summarized
- A new container loads from that summary
- In offline mode, we store the blobs on load that we have. 
- Because some of the blobs are not there, the loader makes a network call for each blob.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		"Fluid.Container.UseLoadingGroupIdForSnapshotFetch2": true,
+		"Fluid.Container.enableOfflineLoad": true,
+	});
+
+	let provider: ITestObjectProvider;
+	const testSnapshotCache = new TestSnapshotCache();
+
+	beforeEach("setup", async function () {
+		provider = getTestObjectProvider({ persistedCache: testSnapshotCache });
+		if (provider.driver.type === "odsp") {
```
Comment (jatgarg) [inline packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts:42]: This should be inverted?

File: packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		const dataStoreB = await containerRuntime.createDataStore(
+			TestDataObjectType,
+			loadingGroupId,
+		);
+
+		mainObject._root.set("dataObjectA", dataStoreA.entryPoint);
+		mainObject._root.set("dataObjectB", dataStoreB.entryPoint);
+	};
+
+	it("Should not make odsp network calls", async () => {
```
Comment (jatgarg) [inline packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts:65]: Also, please skip this as we know it is not passing. I will enable it with my PR.

File: packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts (hunk 1 above)
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts:42]: Yes

File: packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts (hunk 2 above)
Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdBlobReads.spec.ts:65]: OK sounds good

**Review by jatgarg — APPROVED** [review]

**Pull request merged**